tests/coverage/CMakeLists.txt

```diff
@@ -0,0 +1,2 @@
+zephyr_sources_ifdef(CONFIG_COVERAGE_HW_TEST coverage.c)
```

Comment:
where is this defined? needs to be in the same commit.

Also, please change the name, I would just call it CONFIG_GCOV